test/parallel/test-fs-utimes-y2K38.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+const common = require('../common');
+const tmpdir = require('../common/tmpdir');
+
+const assert = require('assert');
+const fs = require('fs');
+
+tmpdir.refresh();
+
+// Check for Y2K38 support. For Windows, assume it's there. Windows doesn't have
+// `touch`.
+if (!common.isWindows) {
+  const testFilePath = `${tmpdir.path}/y2k38-test`;
+  const testFileDate = '204001020304';
+  const { spawnSync } = require('child_process');
+  const { status } = spawnSync('touch',
+                               ['-t', testFileDate, testFilePath],
+                               { encoding: 'utf8' });
+  if (status !== 0) {
```

Comment:
The `touch` doesn't fail on my system. It just silently sets the date to Jan 19 2038

Reply:
Ah, OK. It fails on SmartOS and Raspberry Pi devices in CI so I was hoping that was a universal indicator of non-support for Y2K38. But apparently not. I'll restore the previous logic that included `date -r` to confirm.

Reply:
Lots of trial-and-error here to try to get it as close to right as we can. Thanks for your patience with this all.